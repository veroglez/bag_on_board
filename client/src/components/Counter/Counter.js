import React from 'react';
import { Button } from 'components';
import { useStateValue } from 'common/context/contextHooks';
import style from './Counter.module.scss';

const managePassenger = (dispatch, { form }, action) => {
  const counterBags = action === '+' ? form.bags + 1 : form.bags - 1;
  dispatch({
    type: 'update',
    payload: {
      form: { ...form, bags: counterBags },
    },
  });
};

const Counter = () => {
  const [state, dispatch] = useStateValue();
  const { l10n, form } = state;

  return (
    <div className={style.container}>
      <Button
        id="buttonMinus"
        text={l10n.MINUS}
        onClick={() => form.bags >= 0 && managePassenger(dispatch, state, '-')}
        disabled={form.bags === 0}
      />
      <p className={style.text}>{form.bags}</p>
      <Button
        id="buttonPlus"
        text={l10n.PLUS}
        onClick={() => managePassenger(dispatch, state, '+')}
        disabled={form.bags >= 5}
      />
    </div>
  );
};

export default Counter;
