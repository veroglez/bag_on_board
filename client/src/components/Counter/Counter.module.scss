.container {
  display: flex;
  align-items: center;
}
.text {
  min-width: 40px;
  text-align: center;
  margin-bottom: 0;
}