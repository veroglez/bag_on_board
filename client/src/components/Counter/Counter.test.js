
import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import * as Context from 'common/context/contextHooks';
import Counter from './Counter';

Enzyme.configure({ adapter: new Adapter() });

const STATE = {
  l10n: { PLUS: '+', MINUS: '-' },
  form: { bags: 1 },
};

const HOC = () => (
  <Context.StateContext.Provider value={[STATE]}>
    <Counter />
  </Context.StateContext.Provider>
);

describe('<Counter>', () => {
  it('render', () => {
    const tree = renderer.create(<HOC />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('dispath is called with', async () => {
    const dispatch = jest.fn();
    const stateUpdated = {
      type: 'update',
      payload: { form: { bags: 2 } },
    };

    jest
      .spyOn(Context, 'useStateValue')
      .mockImplementation(() => [STATE, dispatch]);
    const wrapper = mount(<HOC />);
    wrapper.find('#buttonPlus').at(0).props().onClick();

    expect(dispatch).toHaveBeenCalledWith(stateUpdated);
  });
});
