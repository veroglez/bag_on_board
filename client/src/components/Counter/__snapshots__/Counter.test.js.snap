// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter> render 1`] = `
<div
  className="container"
>
  <div
    className="content"
  >
    <button
      className="button"
      disabled={false}
      id="buttonMinus"
      onClick={[Function]}
      type="submit"
    >
      <span>
        -
      </span>
    </button>
    <input
      className="counter"
      onChange={[Function]}
      type="number"
      value={0}
    />
    <button
      className="button"
      disabled={false}
      id="buttonPlus"
      onClick={[Function]}
      type="submit"
    >
      <span>
        +
      </span>
    </button>
  </div>
</div>
`;
